tools/test-my-pr.py

```diff
@@ -0,0 +1,116 @@
+
+
+import donate_cpu_lib as lib
+import argparse
+import os
+import sys
+import random
+import time
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description='Run this script from your branch with Cppcheck patch to check to verify your patch against current master. It will compare output of testing bunch of opensource packages')
+    parser.add_argument('-j', default=1, type=int, help='Concurency execution threads')
+    parser.add_argument('-p', default=1000, type=int, help='Count of packages to check')
+    parser.add_argument('-o', default='my_check_diff.log', help='Filename of result inside a working path dir')
+    parser.add_argument('--work-path', '--work-path=', default=lib.work_path, type=str, help='Working directory for reference repo')
+    args = parser.parse_args()
+
+    print(args)
+
+    work_path = args.work_path
+    if not os.path.exists(work_path):
+        os.mkdir(work_path)
+    cppcheck_path = os.path.join(work_path, 'cppcheck')
+
+    jobs = '-j' + str(args.j)
+    result_file = os.path.join(work_path, args.o)
+    my_root = os.path.dirname(os.path.dirname(os.path.abspath(sys.argv[0])))
+
+    if not lib.get_cppcheck(cppcheck_path, work_path):
+        print('Failed to clone clear version of Cppcheck, retry later')
+        sys.exit(1)
+
+    if not lib.compile(cppcheck_path, jobs):
+        print('Failed to compile clear version of Cppcheck')
+        sys.exit(1)
+
+    print('Testing your PR from root: ' + my_root)
```

Comment:
I think `root` is an unfortunate word. We are not talking about `/`. Maybe write `branch` instead or lookup the actual branch name.

Reply:
Completely agree, root is not the best word. Also I suppose to rephrase some texts today.